test/parser/postgresparser_test.cpp

```diff
@@ -0,0 +1,273 @@
+//===----------------------------------------------------------------------===//
+//
+//                         Peloton
+//
+// parser_test.cpp
+//
+// Identification: test/parser/parser_test.cpp
+//
+// Copyright (c) 2015-16, Carnegie Mellon University Database Group
+//
+//===----------------------------------------------------------------------===//
+
+#include <string>
+#include <vector>
+
+#include "common/harness.h"
+#include "common/macros.h"
+#include "common/logger.h"
+#include "parser/postgresparser.h"
+#include "parser/parser.h"
+
+namespace peloton {
+namespace test {
+
+//===--------------------------------------------------------------------===//
+// PostgresParser Tests
+//===--------------------------------------------------------------------===//
+
+class PostgresParserTests : public PelotonTest {};
+
+TEST_F(PostgresParserTests, BasicTest) {
+  std::vector<std::string> queries;
+
+  // Simple select
+  queries.push_back("SELECT * FROM foo;");
+
+  auto parser = parser::PostgresParser::GetInstance();
+  // auto ref_parser = parser::Parser::GetInstance();
+  // Parsing
+  UNUSED_ATTRIBUTE int ii = 0;
+  for (auto query : queries) {
+    auto stmt_list = parser.BuildParseTree(query).release();
+    // auto ref_stmt_list = ref_parser.BuildParseTree(query).release();
+    EXPECT_TRUE(stmt_list->is_valid);
+    if (stmt_list->is_valid == false) {
+      LOG_ERROR("Message: %s, line: %d, col: %d", stmt_list->parser_msg,
+                stmt_list->error_line, stmt_list->error_col);
+    }
+    // LOG_TRACE("%d : %s", ++ii, stmt_list->GetInfo().c_str());
+    LOG_INFO("%d : %s", ++ii, stmt_list->GetInfo().c_str());
+    // LOG_INFO("%d : %s", ++ii, ref_stmt_list->GetInfo().c_str());
+    delete stmt_list;
+    // delete ref_stmt_list;
+  }
+}
+
+TEST_F(PostgresParserTests, AggTest) {
+  std::vector<std::string> queries;
+
+  // Select with functional call
+  queries.push_back("SELECT COUNT(*) FROM foo;");
+  queries.push_back("SELECT COUNT(DISTINCT id) FROM foo;");
+  queries.push_back("SELECT MAX(*) FROM foo;");
+  queries.push_back("SELECT MIN(*) FROM foo;");
+
+  auto parser = parser::PostgresParser::GetInstance();
+  // auto ref_parser = parser::Parser::GetInstance();
```

Comment:
I'm keeping those for the time being since sometimes I still want to compare with the Peloton parser to check my transformation. I can submit a version without the references but keep those locally, do you prefer that?